Cards/Effects/EffectDamage.gd

```diff
@@ -3,10 +3,12 @@ class_name EffectDamage extends EffectBase
 # @Override
 func apply_effect(caster: Entity, target: Entity, value: int) -> void:
 	var target_damage_data: DealDamageData = DealDamageData.new()
-	target_damage_data.damage = value
-	target_damage_data.caster = caster
+	# calculate modified damage given caster and target stats
+
+	target_damage_data.damage = calculate_value("damage",caster, target, value)
+	target_damage_data.caster = caster # is this useful now that we calculate the stats before anyway ?
```

Comment:
I don't think we need `DealDamageData` any do we? We have access to the health via the Entity object, and since we make different Effects that does unique things EffectDamage should have the logic to subtract the health.

Reply:
Yeah I was thinking the same. Just wanted to make sure. `DealDamageData` was useful when some logic was left to the `deal_damage` function, but now it's treated as any other effect.

Reply:
Maybe this should be done at another time ? because there are a lot of places that have `DealDamageData` (like the tests for example).
On the other hand, the more we wait, the more things are built on type of it and the more changes will need to be made.

Reply:
I'll let you make the final say if you want to refactor this out now or not! If we don't let's make an issue to clean this up.

Reply:
I think it's out of the scope of the PR, because I need to modify more than stats and status
I might take said issue myself, but I will make an issue for it i think

Reply:
I'll go ahead and resolve!

Reply:
I'm just keeping this open until i make the issue